File: CivitasRuby/Sorpresa_JugadorEspeculador.rb
module Civitas
    class Sorpresa_JugadorEspeculador < Sorpresa
        public_class_method :new
        def initialize(fianza,texto)
            super(texto)
            @fianza=fianza
        end

        def aplicarAJugador(actual,todos)
            if jugadorCorrecto(actual,todos)
                informe(actual, todos)
                aux=JugadorEspeculador.nuevoEspeculador(todos[actual],@fianza)
                puts aux.nombre
                todos[actual]=aux

            end
        end

        def toString

            return @texto

        end
    end
end

File: CivitasJava/civitas/Sorpresa_JugadorEspeculador.java
package civitas;

import java.util.ArrayList;

public class Sorpresa_JugadorEspeculador extends Sorpresa
{
    private int fianza;

    Sorpresa_JugadorEspeculador(int fianza,String texto)
    {
        super(texto);
        this.fianza=fianza;
        
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {   
            super.informe(actual, todos);
            todos.set(actual, new JugadorEspeculador(todos.get(actual),this.fianza));
        }
    }

    @Override
    public String toString()
    {
        return this.texto;
    }
}